library/bind.c

```diff
@@ -453,6 +465,7 @@ static void remove_binding(struct binding_s *b) {
 void on_unbind(void *ctx, message *m, int code) {
     struct binding_s *b = ctx;
     struct ziti_conn *conn = b->conn;
+    b->waiter = NULL;
```

Comment:
did this waiter get removed from `b->ch->waiters`?

Reply:
yes, once you get a reply callback the waiter is no longer valid